immuneML/reports/data_reports/PDBDataReport.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+from immuneML.reports.encoding_reports.EncodingReport import EncodingReport
+from plotly.subplots import make_subplots
+
+import pandas as pd
+import plotly.express as px
+from Bio.PDB import PDBParser
+
+from immuneML.data_model.dataset.RepertoireDataset import RepertoireDataset
+from immuneML.data_model.repertoire.Repertoire import Repertoire
+from immuneML.reports.ReportOutput import ReportOutput
+from immuneML.reports.ReportResult import ReportResult
+from immuneML.reports.data_reports.DataReport import DataReport
+from immuneML.util.PathBuilder import PathBuilder
+
+
+class PDBDataReport(EncodingReport):
+    """
+    Generates a histogram of the lengths of the sequences in a RepertoireDataset.
+
+    YAML specification:
+
+    .. indent with spaces
+    .. code-block:: yaml
+
+        my_sld_report: NewSequenceLengthDistribution
+
+    """
+
+    @classmethod
+    def build_object(cls,
+                     **kwargs):  # called when parsing YAML - all checks for parameters (if any) should be in this function
+        return PDBDataReport(**kwargs)
+
+    def __init__(self, dataset: RepertoireDataset = None, batch_size: int = 1, result_path: Path = None,
+                 name: str = None, randomInt: int = 0):
+        super().__init__(dataset=dataset, result_path=result_path, name=name)
+        self.randomInt = randomInt
+        self.batch_size = batch_size
+
+    def check_prerequisites(
+            self):  # called at runtime to check if the report can be run with params assigned at runtime (e.g., dataset is set at runtime)
+
+        return True
+
+    def _generate(self) -> ReportResult:  # the function that creates the report
+
+        dataset = np.array(self.dataset.encoded_data.examples.copy(),dtype=float)
+        output_files = []
+
+
+        for i in range(0, len(dataset)):
+
+                fileName = Path(self.dataset.get_example_ids()[i])
+                PathBuilder.build(self.result_path)
```

Comment:
PathBuilder.build(self.result_path) is called inside the loop now, but should probably be outside? because self.result_path shouldn't change

Reply:
Yes, you're right. For some reason I though that each structure had their own path